### PR TITLE
Update to matchms 0.4.0

### PR DESCRIPTION
**Pull request by florian-huber**
- Changed environment from ``matchms 0.3.4`` to ``matchms 0.4.0``
- Update integration test to be closer to our current "default" pipeline.

### CODE REVIEW

File: tests/test_calc_vector.py [hunk 1]
```diff
@@ -39,20 +39,21 @@ def test_calc_vector_higher_than_allowed_missing_percentage():
...
-    expected_vector = numpy.array([-0.0760859, -0.09670882, 0.10740811, 0.04696679, 0.00893024,
-                                   -0.04496526, -0.0079267, -0.05153507, -0.00433566, -0.06211582])
-    assert document.words[1] not in model.wv.vocab, "Expected word to be missing from given model."
+    expected_vector = numpy.array([0.12775915, -1.17673617, -0.14598507, -0.40189132, 0.36908966,
+                                   1.11608575, -1.46774333, -0.31442554, -0.23168877, -0.29420064])
+    assert document.words[0] not in model.wv.vocab, "Expected word to be missing from given model."
     assert numpy.all(vector == pytest.approx(expected_vector, 1e-5)), "Expected different document vector."
 
 
 def import_pretrained_model():
```
Comment (sverhoeven) [inline tests/test_calc_vector.py:55]: Use [pytest fixture](https://docs.pytest.org/en/latest/fixture.html) instead.

```py
import pytest

def test_calc_vector_higher_than_allowed_missing_percentage(model):
    ...

@pytest.fixture(scope="function")
def model():
    ...
    return gensim.models.Word2Vec.load(model_file)
```

**Review by sverhoeven — APPROVED** [review]
Looks OK to me.

Nice to have documented tests. 

I only have a small optional improvement.

File: tests/test_calc_vector.py (hunk 1 above)
Comment (florian-huber) [reply tests/test_calc_vector.py:55]: I tried it but then ran into ``AttributeError: 'function' object has no attribute 'wv'``. So I would (for time sake) leave it as it is if that is OK?

**Comment by florian-huber** [comment]
Thanks a lot for reviewing Stefan!

**Pull request merged**